clients/pkg/promtail/targets/manager.go

```diff
@@ -232,6 +240,16 @@ func NewTargetManagers(
 				return nil, errors.Wrap(err, "failed to make cloudflare target manager")
 			}
 			targetManagers = append(targetManagers, cfTargetManager)
+		case S3Configs:
+			pos, err := getPositionFile()
+			if err != nil {
+				return nil, err
+			}
+			objectStoreTargetManager, err := objectstore.NewTargetManager(objectStoreMetrics, logger, pos, client, scrapeConfigs)
```

Comment:
If we want to support more cloud provider then we should do `NewS3TargetManager`.

If not then we should rename the package to be `s3.`

Reply:
Have the check for what type of config it is here https://github.com/adityacs/loki/blob/object_storage_scrape/clients/pkg/promtail/targets/objectstore/targetmanager.go#L58

So, it should be okay to have just `NewTargetManager`